src/main/java/service/ActivityStrategyImpl.java

```diff
@@ -0,0 +1,23 @@
+package service;
+
+import java.util.Map;
+import service.activityhandler.ActivityHandler;
+
+public class ActivityStrategyImpl implements ActivityStrategy {
+    private final Map<String, ActivityHandler> activityHandlerMap;
+
+    public ActivityStrategyImpl(Map<String, ActivityHandler> activityHandlerMap) {
+        this.activityHandlerMap = activityHandlerMap;
+    }
+
+    @Override
+    public ActivityHandler get(String activityType) {
+
+        if (activityType.equals("b") || activityType.equals("p")
```

Comment:
that violates O from SOLID, instead use: `if (!map.containsKey(activityCode)) { throw an exception }`